server/src/main/java/org/opensearch/transport/InboundPipeline.java

```diff
@@ -90,12 +89,13 @@ public InboundPipeline(
         LongSupplier relativeTimeInMillis,
         InboundDecoder decoder,
         InboundAggregator aggregator,
-        BiConsumer<TcpChannel, InboundMessage> messageHandler
+        BiConsumer<TcpChannel, ProtocolInboundMessage> messageHandler
     ) {
         this.relativeTimeInMillis = relativeTimeInMillis;
         this.statsTracker = statsTracker;
         this.decoder = decoder;
         this.aggregator = aggregator;
+        this.protocolBytesHandlers = List.of(new NativeInboundBytesHandler(isClosed, pending, decoder, aggregator, statsTracker));
```

Comment:
Please remove the `isClosed`, we need to add `private boolean isClosed = false;` to `NativeInboundBytesHandler` and properly set it when `currentHandler.close();` is called.